CHANGELOG.next.toml

```diff
@@ -52,3 +52,40 @@ message = "Fix bug that can cause panics in paginators"
 references = ["smithy-rs#1903", "smithy-rs#1902"]
 meta = { "breaking" = false, "tada" = false, "bug" = true, "target" = "client"}
 author = "rcoh"
+
+[[aws-sdk-rust]]
+message = """
+The HTTP connector used when making requests is now configurable. This is especially useful when testing:
+
+```rust
+    let conn = TestConnection::new(vec![
+        // tuples of requests and responses
+    ]);
```

Comment:
Should we use a more real world example for the connector replacement? Perhaps for changing the trust store or something? `TestConnection` isn't really well documented or ready for use by customers.

This webpki roots example on the hyper adapter might be a good one to use:
https://github.com/awslabs/smithy-rs/blob/3e8fc306ff938c84964ea01160311d70495a1f18/rust-runtime/aws-smithy-client/src/hyper_ext.rs#L60-L73